### ISSUE TITLE
SDN 4.0.x (unreleased) not yet compatible with Spring Data Rest 3.x

### ISSUE THREAD

**Issue by espiegelberg** [spring-projects/spring-data-neo4j#307]
I am working on an existing projected based on Spring Data Neo4j (SDN) 3.x and Spring Data Rest (SDR) 3.x. Having begun upgrading to make use of the soon to be released SDN by using 4.0.0-BUILD-SNAPSHOTs, I have uncovered that SDN 4 is not yet compatible with SDR 3.

Recognizing that SDN 4 is not yet released, is there a general timeline for when SDN4 and SDR will become compatible?

**Comment by espiegelberg** [comment]
To determine if the issue was our application or something in SDN or SDNR, I added the latest SDNR dependencies to the latest snd-university sample project. The result was the exact same warning log messages and 500 errors as we are seeing from our application. This strengthens the indication that the issue lies in SDNRest.

Here are the steps used to reproduce the issue:

1)      Get the latest sdn-university code
2)      Modify the pom.xml to include the following two dependencies:

                        <dependency>
                                    <groupId>org.springframework.boot</groupId>
                                    <artifactId>spring-boot-starter-data-rest</artifactId>
                        </dependency>
                        <dependency>
                                    <groupId>org.springframework.data</groupId>
                                    <artifactId>spring-data-neo4j-rest</artifactId>
                                    <version>3.3.1.RELEASE</version>
                        </dependency>

3)      Start a local instance of Neo4j (we’re using v2.2.3) with an empty database
4)      Rebuild and start sdn-university with “mvn clean spring-boot:run”
5)      Using your browser, hit "http://localhost.capella.edu:8080/students
"; you will get a valid and correct RESTful response from SDNRest with no students. This proves that SDNRest is functional and responding to request against the Student entity
6)      Using your browser, hit "http://localhost:8080/api/reload
“; this populates the sdn-university sample data, including creation of 25 students
7)      Using your browser, once again hit "http://localhost.capella.edu:8080/students
“; your browser results in a blank screen (on Chrome) due to a 500 error being returned

When executing step #7 above, the following info is logged to the console:
…
14:06:27.914 [qtp1850410949-20] INFO  org.neo4j.ogm.session.Neo4jSession - --------- new request ----------
14:06:27.914 [qtp1850410949-20] INFO  org.neo4j.ogm.session.Neo4jSession - getOrCreateTransaction() being called on thread: 20
14:06:27.914 [qtp1850410949-20] INFO  org.neo4j.ogm.session.Neo4jSession - Session identity: org.neo4j.ogm.session.delegates.TransactionsDelegate@19d98a9d
14:06:27.914 [qtp1850410949-20] INFO  org.neo4j.ogm.session.Neo4jSession - There is no existing transaction, creating a transient one
14:06:27.915 [qtp1850410949-20] INFO  o.n.o.session.request.DefaultRequest - POST http://localhost:7474/db/data/transaction/commit, request: {"statements":[{"statement":"MATCH (n:`Student`) WITH n SKIP 0 LIMIT 20 MATCH p=(n)-[*0..1]-(m) RETURN collect(distinct p)","parameters":{},"resultDataContents":["graph"],"includeStats":false}]}
14:06:28.038 [qtp1850410949-20] INFO  o.n.o.session.request.DefaultRequest - Response is OK, creating response handler
14:06:28.245 [qtp1850410949-20] INFO  o.n.o.session.response.JsonResponse - Closing HttpResponse
14:06:28.246 [qtp1850410949-20] INFO  o.n.o.session.response.JsonResponse - Closing HttpResponse
14:06:28.248 [qtp1850410949-20] WARN  o.s.d.n.m.Neo4jPersistentEntity - [entity].getPropertyAccessor(Student{id=16, name='Portia Vasquez', courses=0, studyBuddies=0}) called but not implemented
14:06:28.252 [qtp1850410949-20] WARN  o.s.d.n.m.Neo4jPersistentEntity - [entity].getIdentifierAccessor(Student{id=16, name='Portia Vasquez', courses=0, studyBuddies=0}) called but not implemented
…

These same results – getting a valid response when there are no instances in the database but getting a 500 error when an instance exists – can be reproduced for any entity (ie: school, departments, etc).

I have copy and pasted the generated Cypher query from above into the Neo4j console and verified it executes correctly. I think the issue is related to the two WARN messages in the last two lines of the console output.

To take both sdn-university and our application out of the equation, I have also tested using Chrome’s Advanced REST Client to send GET and POST requests to read and create data using Spring Data Neo4j Rest endpoints directly. While the POST and PUT do successfully create data in the Neo4j instance, all of these requests (GET, POST, PUT) return 500 errors.
